### PR TITLE
Closes #183 add is domain for ada

### LINKED ISSUE #183: Feature Request:  Add IS domain for antibody data
### Feature Idea

Add IS domain for antibody data. 

This could be used to develop template for ADAB

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by Kcdahn**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately 
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by Fanny-Gautier — CHANGES_REQUESTED** [review]
Please see below comments to addressed. Thanks

File: NEWS.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## New Features
 
+- Added Anti-Drug Antibody specific SDTM dataset: (#183)
+
+    - `is_ada`
```
[the remark this answers is not in the document]
Comment (Kcdahn) [reply NEWS.md]: Corrected.

File: NEWS.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## New Features
 
+- Added Anti-Drug Antibody specific SDTM dataset: (#183)
```
[the remark this answers is not in the document]
Comment (Kcdahn) [reply NEWS.md]: Will update.  I need to first pull Main, then do this and a few other structure updates.

File: pharmaversesdtm.Rproj [hunk 3]
```diff
@@ -1,4 +1,5 @@
 Version: 1.0
+ProjectId: f9a37634-c2df-4716-badb-dd71e07cae25
```
[the remark this answers is not in the document]
Comment (Kcdahn) [reply pharmaversesdtm.Rproj:2]: Sorry about that. I must accidentally did that. I put it back to match main.  Please check.

File: data-raw/is_ada.R [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+      TRUE ~ ISSTRESC
+    ),
+  ) %>%
+  filter(DROPIT == FALSE) %>%
+  group_by(STUDYID, USUBJID) %>%
+  dplyr::mutate(ISSEQ = row_number()) %>%
+  ungroup() %>%
+  arrange(STUDYID, USUBJID, ISTESTCD, ISBDAGNT, VISITDY)
+
+# Compute NAB data --------------------------------------------------------
```
Comment (Fanny-Gautier) [inline data-raw/is_ada.R:184]: Please follow the formatting of comments according to the [tidyverse](https://style.tidyverse.org/syntax.html#comments) style guide, and as described in the [programming strategy](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html).

**Review by Lina2689 — APPROVED** [review]

**Review by Fanny-Gautier — COMMENTED** [review]
Looks good.

**Review by Lina2689 — APPROVED** [review]

File: inst/extdata/sdtms-specs.json [hunk 5]
```diff
@@ -340,6 +340,11 @@
      "source": "Constructed by {admiralmetabolic} developers"
     },
     {
+     "name": "is_ada",
+     "label": "Immunogenicity Specimen Assessments for ADA",
+     "description": "A SDTM IS dataset containing relevant antidrug antibody assessment studies",
+     "source": "Generated dataset",
+     "author": "Kristin Dahnert"
```
[the remark this answers is not in the document]
Comment (bms63) [reply inst/extdata/sdtms-specs.json]: Is this the last update for this PR - can we merge in so the templates work for ADAB?

File: data-raw/is_ada.R [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+# Description: Create IS SDTM dataset for ADA analysis
+#
+# Includes ISBDAGNT and ISTESTCD/ISTEST assigned for SDTMIG v3.4.
+# To use with SDTMIG v3.3, could use ISBDAGNT to assign to ISTESTCD/ISTEST
+#
+# PLACEBO subjects purposely only have BASELINE data.
+# Creates a mix of Baseline pos/neg, Post-Baseline pos/neg, plus some missing visits
+
+# Load libraries ----
+library(tibble)
```
[the remark this answers is not in the document]
Comment (Kcdahn) [reply data-raw/is_ada.R]: Agreed and updated.
